File: portfolio_back/src/main/java/com/visiplus/portfolio/models/Project.java
package com.visiplus.portfolio.models;

import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true)
    private String slug;

    private String title;

    @Lob
    private String description;

    private String startDate;
    private String endDate;

    @ElementCollection
    private List<String> technologies;

    @ElementCollection
    private List<String> tags;

    @ManyToMany(cascade = { CascadeType.MERGE })
    @JoinTable(
            name = "project_skill",
            joinColumns = @JoinColumn(name = "project_id"),
            inverseJoinColumns = @JoinColumn(name = "skill_id")
    )
    private Set<Skill> skills = new HashSet<>() ;

    private String category;

    private String thumbnailUrl;

    @ElementCollection
    private List<String> gallery;

    private String gitUrl;
    private String liveUrl;

    private String role;

    private boolean isFeatured;
    private boolean active;
    private boolean old;

    @Lob
    private String content;

    @Lob
    private String longDescription;
}

File: portfolio_front/src/app/interfaces/project.ts
export interface ProjectResponse{
    id: number;
    slug: string;
    title: string;
    description: string;

    startDate: string;  // format: 'YYYY-MM-DD' attendu
    endDate: string;

    technologies: string[];
    tags: string[];
    category: string;

    thumbnailUrl: string;
    gallery: string[];

    gitUrl: string | null;
    liveUrl: string | null;

    role: string;

    isFeatured: boolean;
    active: boolean;
    old: boolean;

    content: string;  // contenu en Markdown

}
  